Lib/test/test_poplib.py

```diff
@@ -306,9 +307,29 @@ def test_noop(self):
     def test_rpop(self):
         self.assertOK(self.client.rpop('foo'))
 
-    def test_apop(self):
+    def test_apop_normal(self):
         self.assertOK(self.client.apop('foo', 'dummypassword'))
 
+    def test_apop_REDOS(self):
+        old_welcome = self.client.welcome
+
+        # Replace welcome with very long evil welcome.
+        # NB The upper bound on welcome length is currently 2048.
+        # At this length, evil input makes each apop call take
+        # on the order of milliseconds instead of microseconds.
+        evil_welcome = bytes('+OK' + ('<' * 500000), self.client.encoding)
+        self.client.welcome = evil_welcome
```

Comment:
You can use a special helper `swap_attr()` for temporary setting the attribute:
```python
with test.support.swap_attr(self.client, 'welcome', evil_welcome):
```

Reply:
Ah, nice, thanks for the tip.